app/views/public/api/v1/portals/_header.html.erb

```diff
@@ -8,34 +7,111 @@
+            <div class="flex flex-row items-center gap-1">
+              <%= render partial: 'icons/sun' %>
+              <span class="text-xs font-medium"><%= I18n.t('public_portal.header.appearance.light') %></span>
+            </div>
+            <% if @theme.present? && @theme == 'light' %>
+              <%= render partial: 'icons/check-mark' %>
+            <% end %>
+          </button>
+          <button onclick="switchTheme('dark')" class="flex flex-row items-center justify-between px-2 py-2 gap-1 stroke-slate-700 dark:stroke-slate-200 hover:stroke-woot-500 dark:hover:stroke-woot-500 text-slate-800 dark:text-slate-100 hover:text-woot-500 dark:hover:text-woot-500">
+            <div class="flex flex-row items-center gap-1">
+              <%= render partial: 'icons/moon' %>
+              <span class="text-xs font-medium"><%= I18n.t('public_portal.header.appearance.dark') %></span>
+            </div>
+            <% if @theme.present? && @theme == 'dark' %>
+              <%= render partial: 'icons/check-mark' %>
+            <% end %>
+          </button>
         </div>
       </div>
+
+      <%# Locale switcher section %>
+      <% if @portal.config["allowed_locales"].length > 1 %>
+        <div class="flex items-center stroke-slate-700 dark:stroke-slate-200 hover:stroke-woot-500 dark:hover:stroke-woot-500 text-slate-800 dark:text-slate-100 hover:text-woot-500 dark:hover:text-woot-500">
+          <div class="flex items-center gap-1 px-1 py-2">
+            <%= render partial: 'icons/globe' %>
+            <select
+              data-portal-slug="<%= @portal.slug %>"
+              class="bg-white dark:bg-slate-900 appearance-none w-24 overflow-hidden text-ellipsis whitespace-nowrap leading-tight font-medium focus:outline-none text-sm focus:shadow-outline locale-switcher cursor-pointer"
+            >
+              <% @portal.config["allowed_locales"].each do |locale| %>
+                <option <%= locale == params[:locale] ? 'selected': '' %> value="<%= locale %>"><%= "#{language_name(locale)} (#{locale})" %></option>
+              <% end %>
+            </select>
+          </div>
+          <div class="pointer-events-none flex items-center px-2 text-slate-800 dark:text-slate-100">
+            <%= render partial: 'icons/chevron-down' %>
+          </div>
+        </div>
+      <% end %>
     </div>
-    <% end %>
   </nav>
 </header>
+<script>
+  const toggleAppearanceDropdown = () => {
+    const dropdown = document.getElementById('appearance-dropdown');
+    dropdown.style.display = (dropdown.style.display === 'none' || !dropdown.style.display) ? 'flex' : 'none';
+  };
+
+  const switchTheme = (theme) => {
+    const bodyDiv = document.querySelector('.antialiased');
```

Comment:
If this is the body we are targeting, we can simply use `document.body`

```suggestion
    const bodyDiv = document.body;
```

Reply:
Renamed the bodyDiv to parentDiv. We are adding the theme class inside the body.

Reply:
In that case, let's use a `id` or a better class name, which is more specific instead of `antialiased`.

Reply:
DOne